models/ed/R/write.configs.ed.R

```diff
@@ -146,21 +146,13 @@ write.config.ED2 <- function(trait.values, settings, run.id, defaults = settings
     ed2in.text <- read_ed2in(filename)
   }
 
-  metstart <- tryCatch(format(as.Date(settings$run$site$met.start), "%Y"), 
-                       error = function(e) settings$run$site$met.start)
-  metend <- tryCatch(format(as.Date(settings$run$site$met.end), "%Y"), 
-                     error = function(e) settings$run$site$met.end)
-  
-
   ed2in.text <- modify_ed2in(
     ed2in.text,
     latitude = as.numeric(settings$run$site$lat),
     longitude = as.numeric(settings$run$site$lon),
     met_driver = settings$run$inputs$met$path,
     start_date = startdate,
     end_date = enddate,
-    MET_START = metstart,
-    MET_END = metend,
```

Comment:
Not clear why met start and end were removed from the code

Reply:
The way it was set here was wrong -- it would try to set the nonexistent tags `MET_START` and `MET_END`, rather than filling in templates like we used to do. This bug was cancelled out by another bug in `write_ed2in` that prevented it from writing non-existing ED2IN tags, but `barebones=true` in this PR resolved that, which is how I discovered this.

All that said, I saw from your Slack conversation with @bcow that people are actually using this tag, so I added it back (with, IMHO, more robust code) in 8481e7351.

Reply:
See code starting here: https://github.com/PecanProject/pecan/pull/2203/files#diff-304b6e96334a5a21db89565447d07a5fR173

Reply:
> Those tags are in every versions of the ED2IN template I checked

We haven't used the old mechanism for setting ED2IN tags based on `gsub` of things like `@MET_START@` since March (https://github.com/PecanProject/pecan/pull/1870). The current mechanism sets ED2IN tags based on the name of the corresponding directive in the ED2IN, and there are no directives called `MET_START` or `MET_END` (the corresponding directives are `METCYC1` and `METCYCF`). So the removed code referenced here would try to create namelist tags called `NL%MET_START` and `NL%MET_END`, which ED does not expect and would cause to crash on startup. _However_, this bug was cancelled out by another, where non-existent ED2IN tags were not actually being written to the ED2IN, which is why this error went undetected for a while.

`proc_met_startend` does effectively the same thing as the old one, but more verbosely and with some additional error checking. The old code did no validation at all of `met.start/end` being valid values -- it would happily accept arbitrary strings or any other nonsense that would be written to ED2IN, without any warning, which would cause ED to crash on initialization.

> years earlier that 1000

The warning pre-year 1000 doesn't actually do anything except mention that the year looks suspicious. I thought this was fair since at least 90% of PEcAn runs are done after 1900. The goal here was to provide a bit more of a warning in case the tag was mis-interpreted or there was some error in the conversion from a date string to a year. In my opinion, it doesn't actively prevent anyone from doing paleo runs, and could help diagnose other potential problems, but if you disagree, I'm happy to remove it.

Why do I prefer `[[` over `$`?

I don't like `$` because, by default, it performs [partial matching](https://adv-r.hadley.nz/subsetting.html#section-1). So, for example, if we don't have a `site` tag but have a tag like `site_soil` or something similar, the current code will silently match that instead. Example:

```r
l <- list(apple = list(red = 5))
l$a$r
# [1] 5
```

That can introduce all kinds of problems that are very difficult to diagnose (and in fact, I have seen places elsewhere in the PEcAn code where this has caused unexpected behavior). OTOH, using `[[` subsetting forces an exact match, which I find to be safer. If the recursive subsetting is confusing, I can change to `settings[["run"]][["site"]][["met.start"]]`.

Reply:
re: pre-1000 A.D.:  Unless somebody has found a workaround recently, ED didn't like 3-digit years when we did the PalEON long runs.  We worked around this by just adding 1000 to the year (so 850 = 1850; 1850 - 2850).

Reply:
@ashiklom do you mean to say that in March you deprecated the use of `@tag@` syntax in the ED2IN altogether??? I don't think that's common knowledge, since it wasn't explicitly discussed in that PR and my memory was that your PRs from around then were about facilitating your RTM runs (I wasn't aware of impacts on the main workflow). AFAIK everyone else has continued to use those tags without realizing they were just wasting their time.

Point taken on partial matching

For the dates, you don't have to drop the warning, but if you moved it to before 850AD you'd move from 90% of the runs to 99.99%

@crollinson I think I ran into similar issues with PEcAn and JULES that I thought I'd fixed by padding with leading zeros (e.g. 850 had to be 0850 otherwise the filename wasn't parsed correctly). Would be good to know if that makes ED happy too or if there's a deeper issue.

Reply:
> you deprecated the use of @tag@ syntax in the ED2IN altogether?

Yes. See changes starting around [here](https://github.com/PecanProject/pecan/pull/1870/files#diff-304b6e96334a5a21db89565447d07a5fR153).

> I don't think that's common knowledge

Happy to take the blame for that one. I didn't think people were creating new template ED2IN files, so I assumed as long as the resulting ED2IN was the same, the change wouldn't negatively affect anyone. To be fair, I did request reviews from you, @serbinsh , and @istfer , who I thought were the three most active ED users, for exactly this reason, and the CHANGELOG entry does mention that [existing PEcAn code has been altered](https://github.com/PecanProject/pecan/pull/1870/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR33) to use the new utilities I created, but I readily admit that both the PR message and that text are vague given the magnitude of the change.

> For the dates, you don't have to drop the warning, but if you moved it to before 850AD you'd move from 90% of the runs to 99.99%

I suppose, but given the potential issues with 3-digit years pointed out by @crollinson, I'm tempted to leave this as is for now, since it provides an additional hint that such years can be problematic (at least until we can definitively sort out that pre-1000 year work without additional modification in the workflow).